### PR TITLE
Add wxRichTextCtrl XRC and value property support

### PR DESCRIPTION
**Pull request by Kvaz1r**
Fix: #618 

I've not added [property maxlength](https://docs.wxwidgets.org/trunk/overview_xrcformat.html#xrc_wxrichtextctrl) because currently [does nothing](https://docs.wxwidgets.org/trunk/classwx_rich_text_ctrl.html#a3ac4c0377a88864857e678e2bf5a7f14).

### CODE REVIEW

File: plugins/additional/additional.cpp [hunk 1]
```diff
@@ -229,13 +229,21 @@ class RichTextCtrlComponent : public ComponentBase
...
+
 		wxRichTextCtrl* richText = new wxRichTextCtrl( 	(wxWindow*)parent,
 															wxID_ANY,
-															wxEmptyString,
+															text,
 															obj->GetPropertyAsPoint(_("pos")),
 															obj->GetPropertyAsSize(_("size")),
 															obj->GetPropertyAsInteger(_("style")) | obj->GetPropertyAsInteger(_("window_style")));
 
+		if(!text.IsEmpty())
```
Comment (sodevel) [inline plugins/additional/additional.cpp:242]: Please don't use wxWidgets API if there is std::string API available, here i would prefer to use `empty()`. Also add a space between the `if` and the parentheses.

File: plugins/additional/additional.cpp [hunk 2]
```diff
@@ -370,14 +378,18 @@ class RichTextCtrlComponent : public ComponentBase
...
 
 	ticpp::Element* ExportToXrc(IObject* obj) override {
-		ObjectToXrcFilter xrc(obj, _("unknown"), obj->GetPropertyAsString(_("name")));
-		//xrc.AddWindowProperties();
+		ObjectToXrcFilter xrc(obj, _("wxRichTextCtrl"), obj->GetPropertyAsString(_("name")));
+		xrc.AddWindowProperties();
+		if(!obj->GetPropertyAsString(_("value")).IsEmpty())
+		{
+			xrc.AddProperty(_("value"), _("value"), XRC_TYPE_TEXT);
+		}
```
Comment (sodevel) [inline plugins/additional/additional.cpp:386]: Better add this property always, even if it's empty, other elements do this as well

File: plugins/additional/additional.cpp [hunk 3]
```diff
@@ -370,14 +378,18 @@ class RichTextCtrlComponent : public ComponentBase
...
 		return xrc.GetXrcObject();
 	}
 
 	ticpp::Element* ImportFromXrc(ticpp::Element* xrcObj) override {
 		XrcToXfbFilter filter(xrcObj, _("wxRichTextCtrl"));
-		filter.AddWindowProperties();
```
Comment (sodevel) [inline plugins/additional/additional.cpp:380]: That looks like a mistake, the standard properties should also be handled

Comment (Kvaz1r) [reply plugins/additional/additional.cpp:380]: Definitely mistake, fixed. Also updated corresponding to comments above.

File: plugins/additional/additional.cpp [hunk 4]
```diff
@@ -229,13 +229,21 @@ class RichTextCtrlComponent : public ComponentBase
...
+
 		wxRichTextCtrl* richText = new wxRichTextCtrl( 	(wxWindow*)parent,
 															wxID_ANY,
-															wxEmptyString,
+															text,
 															obj->GetPropertyAsPoint(_("pos")),
 															obj->GetPropertyAsSize(_("size")),
 															obj->GetPropertyAsInteger(_("style")) | obj->GetPropertyAsInteger(_("window_style")));
 
+		if( !text.empty() )
```
Comment (sodevel) [inline plugins/additional/additional.cpp:242]: There must have been a slight misunderstanding here, i didn't mean to add padding inside the parentheses (like is seen in old code and shouldn't be used in new code) but just add a space after the if, so this should look like this:
```
if (!text.empty())
```

Comment (Kvaz1r) [reply plugins/additional/additional.cpp]: Right, old style a bit confused me. Fixed.

**Pull request merged**